### PR TITLE
VACMS 20206 - allow numbers of more patterns to process as phone numbers in FL results

### PR DESCRIPTION
**Pull request by eselkin**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- Formerly processing of phone numbers for FL results was very rigid and didn't handle many edge cases including somewhat standard ones (especially from PPMS)
- Phone numbers with parentheses from PPMS/CCP would not display
- Modified the phone number parsing strategy... also left presentation to the component
- Sitewide team, We manage FL

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-cms/issues/20206

## Testing done

- Added testing and coverage
- manually tested the failing facilities on PPMS searches in FL

## Screenshots

<details>
<summary>former (problem)</summary>
Screenshot 2025-01-03 at 4 59 12 PM

</details>

<details>
<summary>PR for result</summary>
Screenshot 2025-01-03 at 5 05 20 PM
</details>

## What areas of the site does it impact?

Facility Locator phone number display

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [x] Events are being sent to the appropriate logging solution
- [x] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

Check
`/find-locations/?address=Alamogordo%2C%20nm&bounds%5B%5D%5B%5D=-106.710304&bounds%5B%5D%5B%5D=32.14977&bounds%5B%5D%5B%5D=-105.210304&bounds%5B%5D%5B%5D=33.64977&bounds%5B%5D%5B%5D%5B%5D=-106.710304&bounds%5B%5D%5B%5D%5B%5D=32.14977&bounds%5B%5D%5B%5D%5B%5D=-105.210304&bounds%5B%5D%5B%5D%5B%5D=33.64977&context=Alamogordo%2C%20New%20Mexico%2C%20United%20States&facilityType=provider&latitude=32.89977&longitude=-105.960304&page=1&radius=56&serviceType=253Z00000X&bounds%5B%5D=-106.710304&bounds%5B%5D=32.14977&bounds%5B%5D=-105.210304&bounds%5B%5D=33.64977`

Must be on v-w local deploy with prod vets-api as api endpoint, eg:
`yarn watch --env api=https://api.va.gov`

because RI will use dev PPMS data which is completely different than prod data

### CODE REVIEW

File: src/applications/facility-locator/utils/phoneNumbers.js [hunk 1]
```diff
@@ -14,13 +14,32 @@
...
 
 export const parsePhoneNumber = phone => {
-  const phoneUS = phone.replace(/^1-/, '');
-  const re = /^(\d{3})[ -]*?(\d{3})[ -]*?(\d{4})\s?(\D*)?[ ]?(\d*)?/;
-  const extension = phoneUS.replace(re, '$5').replace(/\D/g, '');
-  const formattedPhoneNumber = extension
-    ? phoneUS.replace(re, '$1-$2-$3 x$5').replace(/x$/, '')
-    : phoneUS.replace(re, '$1-$2-$3');
-  const contact = phoneUS.replace(re, '$1$2$3');
+  const phoneRegex = /^(?:\+?(?<intl>\d{1,}?)[ -.]*)?\(?(?<ac>\d{3})\)?[- .]*(?<pfx>\d{3})[- .]*(?<linenum>\d{4}),?(?: ?e?xt?e?n?s?i?o?n?\.? ?(?<ext>\d*))?$/i;
```
[the remark this answers is not in the document]
Comment (eselkin) [reply src/applications/facility-locator/utils/phoneNumbers.js:17]: Sure, will do. It's similar to the one in content-build but just with named captures to make it more self-explanatory. But it's true, regexs are never really self-explanatory.

[the remark this answers is not in the document]
Comment (eselkin) [reply src/applications/facility-locator/utils/phoneNumbers.js:32]: Added a somewhat long description of the regex

**Review by randimays — APPROVED** [review]
